src/vscode-dts/vscode.proposed.toolProgress.d.ts

```diff
@@ -0,0 +1,25 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+declare module 'vscode' {
+
+	/**
+	 * todo@connor4312: `vscode.window.withProgres` can take this interface as well.
+	 */
+	export interface ProgressStep {
+		/**
+		 * A progress message that represents a chunk of work
+		 */
+		message?: string;
+		/**
+		 * An increment for discrete progress. Increments will be summed up until 100 (100%) is reached
+		 */
+		increment?: number;
+	}
+
+	export interface LanguageModelTool<T> {
+		invoke(options: LanguageModelToolInvocationOptions<T>, token: CancellationToken, progress: Progress<ProgressStep>): ProviderResult<LanguageModelToolResult>;
```

Comment:
Could the progress go in `options`?

Reply:
We talked about this a bit this morning, putting it in the options is not great because `LanguageModelToolInvocationOptions` are also created by an extension in `vscode.lm.invokeTool` and they have no way of creating the right `Progress` object. Of course we could have LangaugeModelTool.invoke take something that `extends` that interface and add Progress in there.